tests/test6.c

```diff
@@ -1,15 +1,27 @@
+
+#define uthash_ptrdiff_t long
+
+#define uthash_noheaders
 #include "uthash.h"
+#include <string.h>
 #include <stdlib.h>   /* malloc */
 #include <stdio.h>    /* printf */
 
 /* Set up macros for alternative malloc/free functions  */
 #undef uthash_malloc
 #undef uthash_free
+#undef uthash_strlen
+#undef uthash_memcmp
 #define uthash_malloc(sz) alt_malloc(sz)
 #define uthash_free(ptr,sz) alt_free(ptr)
+#define uthash_strlen(s) alt_strlen(s)
+#define uthash_memcmp(v1,v2,s) alt_memcmp(v1,v2,s)
+
+static void * search_key = 0;
+static int seen_strlen = 0;
 
 typedef struct example_user_t {
-    int id;
+    char key[3];
```

Comment:
What would go wrong if you didn't change this struct definition (or most of the code below)?  I seem to recall that uthash always uses `memcmp`, even to compare ints.